[Guidelines] Add `EuiTour`

andreadelrio opened the pull request:
### Summary

- Added guidelines to **EuiTour**. 
- Updated **GuideRuleDescription** so that `description` can be a node instead of just a string.

#### Notes:

- I'm using a bit of a different pattern for some of the images here (e.g. there are some that include screenshots of Kibana), let me know if we prefer not to do this.
- Some of the *.svg files that Figma was exporting contained clipPath tags which don't render properly in Chrome. I tried to find a fix but couldn't, that's the reason why there are some *.png files here.

### Checklist

- [x] Check against **all themes** for compatibility in both light and dark modes
- [x] Checked in **mobile**
- [x] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
~- [ ] Props have proper **autodocs** and **[playground toggles](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md#adding-playground-toggles)**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md)**~
~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
~- [ ] Added or updated **[jest](https://github.com/elastic/eui/blob/master/wiki/testing.md) and [cypress](https://github.com/elastic/eui/blob/master/wiki/cypress-testing.md) tests**~
~- [ ] Checked for **breaking changes** and labeled appropriately~
~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
~- [] A **[changelog](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately~

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5350/

cchaos commented:
🙌  This is fantastic! Can you also share your Figma? I have an idea I'd like to try out.

andreadelrio commented:
> 🙌 This is fantastic! Can you also share your Figma? I have an idea I'd like to try out.

@cchaos Sure! Just added you as an editor of my Figma file.

miukimiu commented:
This looks amazing @andreadelrio. 😍 I love the titles and the images. It gave me a lot of ideas to improve the **EuiEmptyPrompt** guidelines.

cchaos commented on src-docs/src/views/tour/guidelines.js:
```diff
@@ -0,0 +1,186 @@
+      description="If users see value in an action they’ll be more likely to engage."
+    >
+      <EuiFlexItem>
+        <EuiPanel
+          color="subdued"
+          paddingSize="l"
+          hasShadow={false}
+          style={{ justifyContent: 'center', display: 'flex' }}
+        >
+          <EuiPanel style={{ maxWidth: 520 }} paddingSize="s">
+            <EuiImage alt="tour useful step" url={image2Example} />
+          </EuiPanel>
+        </EuiPanel>
```
The code looks like:

```suggestion
      <EuiAspectRatio width={2} height={1}>
        <iframe
          title="tour useful step"
          width="800"
          height="450"
          src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2FOTA9yTu1X0gIiJCthgUtPD%2F%255BGuidelines%255D-EuiTour%3Fnode-id%3D152%253A9193"
        />
     </EuiAspectRatio>
```

cchaos replied:
The `EuiAspectRatio` wrapper makes sure it shrinks on smaller screens.

cchaos replied:
Yeah I'd move all your final frames to the Tour page in the Elastic Library. I'd make sure your frames don't have a background and then when you copy/paste the embed code, you'll want to remove the `style` tag that comes with it.

cchaos reviewed (commented):
So I've had this idea about how to better keep Figma and EUI in sync by using the Embed feature of Figma which gives `iframe` code. The result looks like: 

<img width="866" alt="Screen Shot 2021-11-03 at 11 50 45 AM" src="https://user-images.githubusercontent.com/549577/140095121-a9784e6d-9a78-417a-ba6b-a350cf3624b5.png">

**Pros:**
- If you make an edit in Figma, it automatically updates the image. This helps if we make edits to the component without remembering to update the screenshots.
- Aways Hi-def because it's rendering the vector version

**Cons:**
- Figma has decided to add all this extra chrome around the screenshot. It didn't used to exist. I'm hoping they might make that configurable in the future.
- The file needs to be available to the public to `view`. But I was thinking we'd probably just port all these directly into the Elastic UI library instead of keeping theming in sync in a different file.

andreadelrio commented:
@cchaos I added some Figma embeds... I like it! However the loading time is not the best, I found that other Figma users feel the same way. I also don't love the zoom level that gets applied default and I don't see a way to customize that.

I was thinking that these images that I replaced with Figma embeds could be made with code instead! So if we think that's the way to go I'm happy to code them up.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5350/

cchaos commented:
☹️ I agree that it takes a bit of time to load and we'll probably need to fiddle with the aspect ratio dimensions and the frame dimension to have a better zoom level at most window sizes. But I'd really love to try this approach at least for a bit and see how it feels. We can always revisit re-creating them with code later if we're finding this approach troublesome.

You could also try creating a column layout with the text on the left and Figma on the right to make the frame smaller. I'm out tomorrow but if you give me until Friday EOD, I can also play with it some more.

andreadelrio commented:
> ☹️ I agree that it takes a bit of time to load and we'll probably need to fiddle with the aspect ratio dimensions and the frame dimension to have a better zoom level at most window sizes. But I'd really love to try this approach at least for a bit and see how it feels. We can always revisit re-creating them with code later if we're finding this approach troublesome.
> 
> You could also try creating a column layout with the text on the left and Figma on the right to make the frame smaller. I'm out tomorrow but if you give me until Friday EOD, I can also play with it some more.

I found a workaround for the zoom level, I increased the size of the frame that is surrounding each design. I think now the embeds look better.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5350/

andreadelrio commented:
@cchaos thanks for your PR, I merged it. I've moved the block where the UX research tip is a bit further down, hoping that the flow feels better now. I kept the guideline about animation at the very end because I see it like a "nice to have" but let me know if you think we should move the 3-item block to the bottom instead.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5350/

cchaos reviewed (approved):
Makes sense and looks good to me! Thank you so much for adding these. 💟 

_Looks like there's still one leftover unused `tour_4.svg` that can be remove_

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5350/
